compose/ui/ui/src/jsWasmMain/kotlin/androidx/compose/ui/window/SystemThemeObserver.kt

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2023 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.compose.ui.window
+
+import androidx.compose.runtime.State
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.ui.SystemTheme
+import org.w3c.dom.MediaQueryList
+import org.w3c.dom.Window
+
+internal class SystemThemeObserver(window : Window) {
+
+    val currentSystemTheme : State<SystemTheme>
+        get() = _currentSystemTheme
+
+    private val media: MediaQueryList by lazy {
+        window.matchMedia("(prefers-color-scheme: dark)")
+    }
+
+    private val _currentSystemTheme = mutableStateOf(
+        when {
+            !isSupported -> SystemTheme.Unknown
+            media.matches -> SystemTheme.Dark
+            else -> SystemTheme.Light
+        }
+    )
+
+    // supported by all browsers since 2015
+    // https://developer.mozilla.org/en-US/docs/Web/API/Window/matchMedia
+    private val isSupported: Boolean
+        get() = js("window.matchMedia != undefined").unsafeCast<Boolean>()
+
+    init {
+        if (isSupported) {
+            media.addListener {
```

Comment:
```suggestion
        if (isSupported) {
            // it's deprecated, see https://developer.mozilla.org/en-US/docs/Web/API/MediaQueryList/addListener
            media.addListener {
```

Reply:
>In older browsers MediaQueryList did not yet inherit from [EventTarget](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget), so this method was provided as an alias of [EventTarget.addEventListener()](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener). Use addEventListener() instead of addListener() if it is available in the browsers you need to support.

Can we use `addEventListener` and fallback to `addListener` if it is not supported?

Reply:
```kotlin
val mapState = { m: Event ->
    _currentSystemTheme.value = if (m.unsafeCast<MediaQueryList>().matches)
      SystemTheme.Dark else SystemTheme.Light
}

try {
    media.addEventListener("change", EventListener(mapState))
} catch (t: Throwable) {
    media.addListener(mapState)
}
```

Something like that. Will it do the job?

Reply:
thanks, I missed that part. 
according to the browser support table, we can just use `addEventListener` and not care about super old versions. 
Your last snippet looks good!

Reply:
Awesome. Done with this part